### PR TITLE
refactor(condo): DOMA-8840 refetch comments in one tab

### PR DESCRIPTION
**Pull request by nomerdvadcatpyat**
1. Added the `useBroadcastChannel` hook - it is used to send messages to other browser tabs with the same domain, utilizing the `BroadcastChannel` API.
2. Added the `useExecuteWithLock` hook - it is used to execute a function only in one tab, utilizing the `navigator.locks` API.
3. Added the `usePollTicketComments` hook - polling comments only in one tab and sending ticketIds with updated comments to other `/ticket/[id]` tabs. Then, in the tab where the ticket with updated comments is open, comments with files are refetched.

https://github.com/open-condo-software/condo/assets/52532264/ac76cffc-5f52-4241-a07a-4604c395b818

### CODE REVIEW

File: apps/condo/domains/ticket/hooks/usePollTicketComments.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    const { refetch: refetchSyncComments } = TicketComment.useObjects({}, { skip: true })
+
+    const { sendMessage } = useBroadcastChannel<string[]>(BROADCAST_CHANNEL_NAME, async (ticketIdsWithUpdatedComments) => {
+        if (ticketIdsWithUpdatedComments.includes(get(ticket, 'id'))) {
+            await refetchTicketComments()
+        }
+    })
+
+    const pollTicketComments = useCallback(async () => {
+        if (!localStorage) return
```
Comment (sitozzz) [inline apps/condo/domains/ticket/hooks/usePollTicketComments.tsx:31]: what are the reasons for this check?

Comment (nomerdvadcatpyat) [reply apps/condo/domains/ticket/hooks/usePollTicketComments.tsx:31]: for check that it's client side and localStorage is exist

**Review by sitozzz — APPROVED** [review]
Looks pretty good!

File: apps/condo/domains/common/hooks/useBroadcastChannel.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    const messageSender = useRef<BroadcastChannel | null>(null)
+    const messageReceiver = useRef<BroadcastChannel | null>(null)
+
+    useEffect(() => {
+        messageSender.current = new BroadcastChannel(channelName)
+        messageReceiver.current = sendInCurrentTab ? new BroadcastChannel(channelName) : messageSender.current
+
+        messageReceiver.current.onmessage = (event) => onMessageReceived(event.data)
+    }, [channelName, onMessageReceived, sendInCurrentTab])
```
Comment (Alllex202) [inline apps/condo/domains/common/hooks/useBroadcastChannel.tsx:27]: You should close the channel if you want to create a new one or terminate a connection

File: apps/condo/domains/ticket/hooks/usePollTicketComments.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+        const lastSyncAt = localStorage.getItem(LOCAL_STORAGE_SYNC_KEY)
```
Comment (Alllex202) [inline apps/condo/domains/ticket/hooks/usePollTicketComments.tsx:33]: What if `localStorage` is empty?

File: apps/condo/domains/ticket/hooks/usePollTicketComments.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        const updatedAtStatement = lastSyncAt ? { updatedAt_gt: lastSyncAt } : {}
+
+        const result = await refetchSyncComments({
+            where: {
+                ticket: { organization: { id: get(ticket, 'organization.id', null) } },
```
Comment (Alllex202) [inline apps/condo/domains/ticket/hooks/usePollTicketComments.tsx:38]: `cc` has many organizations

File: apps/condo/domains/common/hooks/useExecuteWithLock.tsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+export function useExecuteWithLock (
+    lockName: string,
+    fn: () => void
+) {
+    const releaseLockRef = useRef<(value?) => void>()
+    const fnRef = useRef(fn)
+
+    useEffect(() => {
+        navigator.locks.request(lockName, () => {
```
Comment (Alllex202) [inline apps/condo/domains/common/hooks/useExecuteWithLock.tsx:12]: should we check that the code is running in the browser?

File: apps/condo/domains/common/hooks/useBroadcastChannel.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    onMessageReceived: (message: T) => void,
+    options: UseBroadcastChannelOptions = DEFAULT_OPTIONS
+) {
+    const { sendInCurrentTab } = options
+
+    const messageSender = useRef<BroadcastChannel | null>(null)
+    const messageReceiver = useRef<BroadcastChannel | null>(null)
+
+    useEffect(() => {
+        messageSender.current = new BroadcastChannel(channelName)
```
Comment (Alllex202) [inline apps/condo/domains/common/hooks/useBroadcastChannel.tsx:23]: same^^

File: apps/condo/domains/ticket/hooks/usePollTicketComments.tsx (hunk 4 above)
Comment (nomerdvadcatpyat) [reply apps/condo/domains/ticket/hooks/usePollTicketComments.tsx]: fix it, pass `pollCommentsQuery` here

**Review by Alllex202 — APPROVED** [review]

**Pull request merged**